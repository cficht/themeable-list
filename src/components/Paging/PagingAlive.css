.Paging {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  border: 10px solid rgb(137, 191, 209);
  text-align: center;
  background-color: lightblue;

  button {
    margin: 0px 20px 0px 20px;
    width: 5%;
    font-size: 20px;
    background-color: rgb(137, 191, 209);
    color: white;
  }

  button:disabled {
    margin: 0px 20px 0px 20px;
    width: 5%;
    font-size: 20px;
    background-color: lightgray;
    color: white;
  }
}