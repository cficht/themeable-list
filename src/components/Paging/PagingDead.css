.Paging {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  border: 10px solid rgb(184, 42, 42);
  text-align: center;
  background-color: red;
  color: white;

  button {
    margin: 0px 20px 0px 20px;
    width: 5%;
    font-size: 20px;
    background-color: rgb(184, 42, 42);
    color: white;
  }

  button:disabled {
    margin: 0px 20px 0px 20px;
    width: 5%;
    font-size: 20px;
    background-color: darkred;
    color: white;
  }
}