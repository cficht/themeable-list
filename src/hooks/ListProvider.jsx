import React, { createContext, useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';

const ListContext = createContext();

export const ListProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  const chararcterTest = [
    {
      id: 1,
      name: 'Rick Sanchez',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    },
    {
      id: 2,
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg'
    },
    {
      id: 3,
      name: 'Summer Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg'
    }
  ];

  useEffect(() => {
    setCharacters(chararcterTest);
  }, []);

  return (
    <ListContext.Provider value={{ characters }}>
      {children}
    </ListContext.Provider>
  );
};

ListProvider.propTypes = {
  children: PropTypes.node
};

export const useCharacters = () => {
  const { characters } = useContext(ListContext);
  return characters;
};
