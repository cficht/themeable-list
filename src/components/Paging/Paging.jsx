import React from 'react';
import { usePage, useHandlePage, useCharacters } from '../../hooks/ListProvider';

const Paging = () => {
  const page = usePage();
  const handlePage = useHandlePage();
  const characters = useCharacters();
  
  return (
    <>
      <button onClick={() => handlePage(-1)} disabled={page === 1}>Previous</button>
      <button onClick={() => handlePage(1)} disabled={characters.length < 20}>Next</button>
    </>
  );
};

export default Paging;
