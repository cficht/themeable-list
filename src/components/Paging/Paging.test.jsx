import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';
import { ListProvider } from '../../hooks/ListProvider';

describe('Paging component', () => {
  it('renders Paging', () => {
    const wrapper = shallow(
      <ListProvider>
        <Paging />
      </ListProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
