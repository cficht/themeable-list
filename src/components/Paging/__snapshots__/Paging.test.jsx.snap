// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component renders Paging 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "handlePage": [Function],
      "page": 1,
      "status": "alive",
      "toggle": [Function],
    }
  }
>
  <Paging />
</ContextProvider>
`;
